Apps/Src/ReadCarCAN.c

```diff
@@ -30,22 +31,24 @@ void Task_ReadCarCAN(void *p_arg) {
 
         // Normal task countinues here
 
+        uint32_t canId;
+
         // Check if BPS sent us a message
-        if (CANbus_Read(buffer) == SUCCESS) {
-            long msg = *((long *)(&buffer[0]));
-            if (msg == 0) {
-                car->IsRegenBrakingAllowed = OFF;
+        if (CANbus_Read(&canId, buffer, CAN_NON_BLOCKING) == SUCCESS) {
+            // If charge_enable, set regen flag
+            if (canId == CHARGE_ENABLE) {
+                car->IsRegenBrakingAllowed = (buffer[0] == 0) ? OFF : ON;
+                Contactors_Set(ARRAY, car->IsRegenBrakingAllowed);
+                faultCounter = 0;
             } else {
-                car->IsRegenBrakingAllowed = ON;
+                // If we didn't get a message, something might have gone wrong
+                faultCounter++;
             }
-
-            faultCounter = 0;
-        } else {
-            faultCounter++;
         }
 
         if (faultCounter >= THRESHOLD) {
             car->IsRegenBrakingAllowed = OFF;
+            Contactors_Set(ARRAY, OFF);
         }
```

Comment:
I'm interested to see how often `faultCounter` is incremented when we run the car. One potential issue I see with this is that if we very occasionally miss a CAN message (eg. once every 10 minutes) the car might seem to randomly shut off after a certain period of time. It might make sense to reset `faultCounter` to zero periodically